skribenten-backend/src/main/kotlin/no/nav/pensjon/brev/skribenten/auth/AuthorizeAnsattSakTilgang.kt

```diff
@@ -81,28 +85,43 @@ private fun sjekkAdressebeskyttelse(
 
 private suspend fun sjekkEnhetstilgang(
     navIdent: String,
-    sak: SakSelection,
-    enheterResult: Deferred<ServiceResult<List<NAVEnhet>>>
-): AuthAnsattSakTilgangResponse? =
-    enheterResult.await().map { enheter ->
-        when {
-            erGenerellSakMedEnhet0001(sak) -> null // får tilgang
-            !harTilgangTilSakSinEnhet(enheter, sak) -> {
-                logger.warn("Tilgang til sak ${sak.saksId} avvist for $navIdent: mangler tilgang til enhet ${sak.enhetId}")
-                AuthAnsattSakTilgangResponse("Mangler enhetstilgang til sak", HttpStatusCode.Forbidden)
-            }
-            else -> null // får tilgang
+    penSakTilgangDeferred: Deferred<ServiceResult<PenSakTilgang>>,
+    navansattEnheterDeferred: Deferred<ServiceResult<List<NAVEnhet>>>,
+): AuthAnsattSakTilgangResponse? {
+    val navansattEnheter = navansattEnheterDeferred.await().map { it }
+        .catch { message, httpStatusCode ->
+            logger.error("En feil oppstod ved henting av PenSakTilgang under sjekk av enhetstilgang: $message. Httpstatus: $httpStatusCode")
+            return AuthAnsattSakTilgangResponse("En feil oppstod ved henting av NAVEnheter for ansatt: $navIdent", httpStatusCode)
         }
-    }.catch { msg, status ->
-        logger.error("Kunne ikke henter NAVenheter for ansatt $navIdent: $status - $msg")
-        AuthAnsattSakTilgangResponse("En feil oppstod ved henting av NAVEnheter for ansatt: $navIdent", HttpStatusCode.InternalServerError)
-    }
 
-private fun erGenerellSakMedEnhet0001(sak: SakSelection) =
-    sak.sakType == SakType.GENRL && sak.enhetId == "0001"
+    val penSakTilgang = penSakTilgangDeferred.await().map { it.idForEnheterMedTilgang }
+        .catch { message, httpStatusCode ->
+            logger.error("En feil oppstod ved henting av NAVEnhet under sjekk av enhetstilgang: $message. Httpstatus: $httpStatusCode")
+            return AuthAnsattSakTilgangResponse("En feil oppstod under henting av PEN sakstilganger for ansatt: $navIdent", httpStatusCode)
+        }
+
+    val sakId = penSakTilgangDeferred.await().map { it.saksId }
+
+    return when {
+        !harTilgangTilSakSinEnhet(navansattEnheter, penSakTilgang) -> {
+            logger.warn("Tilgang til sak: $sakId avvist for $navIdent: mangler tilgang til en av følgende: enheter $penSakTilgang")
+            AuthAnsattSakTilgangResponse("Mangler enhetstilgang til sak", HttpStatusCode.Forbidden)
+        }
+        else -> null // får tilgang
+    }
+}
 
-private fun harTilgangTilSakSinEnhet(enheter: List<NAVEnhet>, sak: SakSelection) =
-    enheter.any { it.id == sak.enhetId }
+fun harTilgangTilSakSinEnhet(navAnsattEnheter: List<NAVEnhet>, penSakEnheter: List<String>): Boolean {
+    var tilgang = false
```

Comment:
Denne kan forenkles til:
`penSakEnheter.any { sakEnhet -> navAnsattEnheter.any { sakEnhet == it.id } }`

Reply:
nice